v6.1.0 breaking change on Authentication middleware

Upon upgrading from v6.0.1 to v6.1.0 our project started getting build errors on our generated routes.ts attempting to invoke our authentication middleware.

```
src/routes/routes.ts:1703:83 - error TS2554: Expected 2-3 arguments, but got 4.
expressAuthentication(request, name, secMethod[name], response)
```

## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

Build does not fail in minor version upgrade

## Current Behavior

Build fails due to authentication method signature change.

## Possible Solution

This possibly only affects projects with `strict: true` in the compiler options. It appears the breaking change occurred in https://github.com/lukeautry/tsoa/pull/1557.

## Context (Environment)

Version of the library: 6.1.0
Version of NodeJS: 18.19.1
Version of Typescript: 5.1.3

- Confirm you were using yarn not npm: [ ]

## Detailed Description

<!--- Provide a detailed description of the change or addition you are proposing -->

## Breaking change?

<!--- if this is a breaking change, then please describe how it breaks the existing libraries inputs/outputs -->
<!--- if this is a breaking change, then discuss how we can mitigate that -->

--- a/packages/cli/src/routeGeneration/templates/express.hbs
+++ b/packages/cli/src/routeGeneration/templates/express.hbs
@@ -24,6 +24,11 @@ const multer = require('multer');
 const upload = multer({{{json multerOpts}}});
 {{/if}}
 
+{{#if authenticationModule}}
+const expressAuthenticationRecasted = expressAuthentication as (req: ExRequest, securityName: string, scopes?: string[], res?: ExResponse) => Promise<any>;
+{{/if}}
+
+
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
 
 const models: TsoaRoute.Models = {
@@ -141,7 +146,7 @@ export function RegisterRoutes(app: Router) {
 
                     for (const name in secMethod) {
                         secMethodAndPromises.push(
-                            expressAuthentication(request, name, secMethod[name], response)
+                            expressAuthenticationRecasted(request, name, secMethod[name], response)
                                 .catch(pushAndRethrow)
                         );
                     }
@@ -153,7 +158,7 @@ export function RegisterRoutes(app: Router) {
                 } else {
                     for (const name in secMethod) {
                         secMethodOrPromises.push(
-                            expressAuthentication(request, name, secMethod[name], response)
+                            expressAuthenticationRecasted(request, name, secMethod[name], response)
                                 .catch(pushAndRethrow)
                         );
                     }

--- a/packages/cli/src/routeGeneration/templates/koa.hbs
+++ b/packages/cli/src/routeGeneration/templates/koa.hbs
@@ -14,7 +14,7 @@ import { koaAuthentication } from '{{authenticationModule}}';
 import { iocContainer } from '{{iocModule}}';
 import type { IocContainer, IocContainerFactory } from '@tsoa/runtime';
 {{/if}}
-import type { Context, Next, Middleware } from 'koa';
+import type { Context, Next, Middleware, Request as KRequest, Response as KResponse } from 'koa';
 import type * as KoaRouter from '@koa/router';
 {{#if useFileUploads}}
 {{#if esm}}
@@ -24,6 +24,10 @@ const multer = require('@koa/multer');
 {{/if}}
 const upload = multer({{{json multerOpts}}});
 {{/if}}
+{{#if authenticationModule}}
+const koaAuthenticationRecasted = koaAuthentication as (req: KRequest, securityName: string, scopes?: string[], res?: KResponse) => Promise<any>;
+{{/if}}
+
 
 // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
 
@@ -140,7 +144,7 @@ export function RegisterRoutes(router: KoaRouter) {
 
                     for (const name in secMethod) {
                         secMethodAndPromises.push(
-                            koaAuthentication(context.request, name, secMethod[name], context.response)
+                            koaAuthenticationRecasted(context.request, name, secMethod[name], context.response)
                                 .catch(pushAndRethrow)
                         );
                     }
@@ -150,7 +154,7 @@ export function RegisterRoutes(router: KoaRouter) {
                 } else {
                     for (const name in secMethod) {
                         secMethodOrPromises.push(
-                            koaAuthentication(context.request, name, secMethod[name], context.response)
+                            koaAuthenticationRecasted(context.request, name, secMethod[name], context.response)
                                 .catch(pushAndRethrow)
                         );
                     }

--- a/tests/fixtures/inversify-cpg/authentication.ts
+++ b/tests/fixtures/inversify-cpg/authentication.ts
@@ -1,6 +1,6 @@
 import * as express from 'express';
 
-export function expressAuthentication(req: express.Request, name: string, _scopes: string[] | undefined, res: express.Response): Promise<any> {
+export function expressAuthentication(req: express.Request, name: string, _scopes: string[] | undefined): Promise<any> {
   if (req.query && req.query.tsoa && req.query.tsoa === 'abc123456') {
     return Promise.resolve({});
   } else {

--- a/tests/fixtures/koaNoAdditional/authentication.ts
+++ b/tests/fixtures/koaNoAdditional/authentication.ts
@@ -1,6 +1,6 @@
 import { Request, Response } from 'koa';
 
-export function koaAuthentication(request: Request, name: string, scopes: string[] | undefined, response: Response): Promise<any> {
+export function koaAuthentication(request: Request, name: string, scopes: string[] | undefined): Promise<any> {
   if (name === 'api_key') {
     let token;
     if (request.query && request.query.access_token) {

